pep-0394.txt

```diff
@@ -5,104 +5,167 @@ Last-Modified: $Date$
-  if installed, refers to the same target as ``python2``, unless the user
-  deliberately overrides this or a virtual environment is active.
-* however, end users should be aware that ``python`` refers to ``python3``
-  on at least Arch Linux (that change is what prompted the creation of this
-  PEP), so ``python`` should be used in the shebang line only for scripts
-  that are source compatible with both Python 2 and 3.
-* in preparation for an eventual change in the default version of Python,
-  Python 2 only scripts should either be updated to be source compatible
-  with Python 3 or else to use ``python2`` in the shebang line.
-
+* Distributions can choose whether ``python`` is installed and, if it is,
+  whether it refers to the same target as ``python2`` or ``python3``.
+* End users should be aware that the ``python`` command is inconsistent
+  across Unix-like systems.
+* In activated virtual environments, ``python`` always refers to the
+  environment's Python version.
+* Upstream projects should prefer the ``python3`` shebang even if they support
+  both Python 3 and Python 2.
+* Upstream projects only supporting Python 2 should use the ``python2``
+  shebang.
 
 Recommendation
 ==============
 
-* Unix-like software distributions (including systems like Mac OS X and
+For distributors
+----------------
+
+* Unix-like software distributions (including systems like macOS and
   Cygwin) should install the ``python2`` command into the default path
   whenever a version of the Python 2 interpreter is installed, and the same
   for ``python3`` and the Python 3 interpreter.
 * When  invoked, ``python2`` should run some version of the Python 2
   interpreter, and ``python3`` should run some version of the Python 3
   interpreter.
-* If the ``python`` command is installed, it should invoke the same version of
-  Python as the ``python2`` command (however, note that some distributions
-  have already chosen to have ``python`` implement the ``python3``
-  command; see the `Rationale`_ and `Migration Notes`_ below).
-* The Python 2.x ``idle``, ``pydoc``, and ``python-config`` commands should
-  likewise be available as ``idle2``, ``pydoc2``, and ``python2-config``,
-  with the original commands invoking these versions by default, but possibly
-  invoking the Python 3.x versions instead if configured to do so by the
+* If the ``python`` command is installed, it should either invoke the same
+  version of Python as the ``python3`` command or as the ``python2``
+  command.
+* Distributors may choose to set the behavior of the ``python`` command,
+  not provide it at all, or make it configurable by the user or
   system administrator.
-* In order to tolerate differences across platforms, all new code that needs
+* The Python 3.x ``idle``, ``pydoc``, and ``python-config`` commands should
+  likewise be available as ``idle3``, ``pydoc3``, and ``python3-config``
+  (and similarly for Python 2.x). The original commands should either invoke
```

Comment:
"and similarly for Python 2.x" is a bit ambiguous.  Are you suggesting that if distributors include these tools for Python 2, that they be named `idle2`, `pydoc2`, and `python2-config`?  Perhaps it's better to be explicit here.

Reply:
abf3bf3